src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -134,7 +134,7 @@ private async Task<ExecutionResult> CoreExecuteAsync(ExecutionOptions options)
             if (options.Schema == null)
                 throw new InvalidOperationException("Cannot execute request if no schema is specified");
             if (options.Query == null && options.Document == null)
-                throw new InvalidOperationException("Cannot execute request if no query is specified");
+                return new ExecutionResult { Errors = new ExecutionErrors { new QueryMissingError() } };
```

Comment:
```suggestion
                throw new QueryMissingError();
```
I think that initial idea from @Shane32 was just like that - throw typed error to be able to catch it above and return 400 http code. Moreover, switching from throw to return is a breaking change in behavior.

Reply:
@sungam3r Uhhh....???  I would change this behavior here to return an ExecutionResult as @SlavaUtesinov already posted.  While your idea is somewhat logical, it seems like a make-work project.  Now the server project (and any other middleware usage) has to catch errors thrown from DocumentExecuter and check if it's an ExecutionError (or RequestError) and encapsulate it in an ExecutionResult.  Better just to write the code here in the DocumentExecuter.

We have never considered breaking changes to error conditions as preventing a minor version bump, and this change should have no negative effect on any existing code in the server project or provided samples.  For instance, GraphQL v6 (and GraphQL.AspNetCore3) will still return a 400 error and not even call the DocumentExecuter to begin with.

Rather, we are now "supporting a feature" we did not before -- namely, providing a response when there is no query provided.  Again, while this is breaking in the strictest sense, it should be allowed with a minor version bump.

Finally, as stated previously, situations where `result.Executed == false` should already return 400 according to spec, and does so and will continue to do so.  There is no reason why a missing (`null`) query should be a 400 and an empty (`""`) query should return 200.  If a user were returning 200 for empty (`""`) queries, then it should be acceptable to return a 200 for a null query as well.

Reply:
Note: actually the server v6 as it exists now does not return 400 in the case of validation errors.  That situation is fixed within my PR.  But it does check for null and return 400 regardless of the change to this code.

Reply:
I see the suggestion was merged.  I think this is incorrect and should be reverted.

Reply:
#3138

Reply:
OK